content/4_Gold/Tree_Euler.mdx

```diff
@@ -172,11 +170,237 @@ public static void dfs(int i, int p) {
+          <path transform="translate(1104.5 634.816)" d="M0 0h30v65.367H0z"/>
+        </clipPath>
+      </defs>
+    </g>
+    <path vector-effect="non-scaling-stroke" stroke-width="5" stroke-linecap="square" stroke-miterlimit="3" d="M835 356l-174-63M1058 674l-212-3M1085 615L938 420M1141 610l69-213"/>
+  </g>
+</svg>
+
+</center>
+
+Before the **DFS**, our **st** and **en** arrays are initialized with zeros. In this visualization, the first row represents the node indices, the second row represents the **st** array, and the third represents **en**.
+
+Current timer value: 0
+
+<center>
+<table className="text-center">
+<tbody>
+<tr>
+<td>Node Index</td><td>1</td><td>2</td><td>3</td><td>4</td><td>5</td>
+</tr>
+<tr>
+<td>st[i]</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td>
+</tr>
+<tr>
+<td>en[i]</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td>
+</tr>
+</tbody>
+</table>
+</center>
+
+Since we call `dfs(1, 0)`, our timer starts at `st[1]` and sets it to the current value of 0. Then, we proceed to call `dfs(2, 1)` and `dfs(3, 1)`.
+
+Current timer value: 1
+
+<center>
+<table className="text-center">
+<tbody>
+<tr>
+<td>Node Index</td><td>1</td><td>2</td><td>3</td><td>4</td><td>5</td>
+</tr>
+<tr>
+<td>st[i]</td><td className='bg-green-100 dark:bg-green-700'>0</td><td>0</td><td>0</td><td>0</td><td>0</td>
+</tr>
+<tr>
+<td>en[i]</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td>
+</tr>
+</tbody>
+</table>
+</center>
+
+Now we must resolve `dfs(2, 1)` and `dfs(3, 1)`. It does not matter which order we process these in, so for this example, we start with `dfs(2, 1)`. Since the timer is at `1`, we set `st[2]=1`. However, because `2` has no children, we don't call `dfs`. Instead, we set `en[2]` to 1 because our current timer is 2.
+
+Current timer value: 2
+
+<center>
+<table className="text-center">
+<tbody>
+<tr>
+<td>Node Index</td><td>1</td><td>2</td><td>3</td><td>4</td><td>5</td>
+</tr>
+<tr>
+<td>st[i]</td><td>0</td><td className='bg-green-100 dark:bg-green-700'>1</td><td>0</td><td>0</td><td>0</td>
+</tr>
+<tr>
+<td>en[i]</td><td>0</td><td className='bg-red-100 dark:bg-red-700'>1</td><td>0</td><td>0</td><td>0</td>
+</tr>
+</tbody>
+</table>
+</center>
+
+Now we must resolve `dfs(3, 1)`. Similar to before, we set `st[3]` to the value of the timer, which is 2 in this case. Then, we proceed to make the calls `dfs(4, 3)` and `dfs(5, 3)`.
+
+Current timer value: 3
+
+<center>
+<table className="text-center">
+<tbody>
+<tr>
+<td>Node Index</td><td>1</td><td>2</td><td>3</td><td>4</td><td>5</td>
+</tr>
+<tr>
+<td>st[i]</td><td>0</td><td>1</td><td className='bg-green-100 dark:bg-green-700'>2</td><td>0</td><td>0</td>
+</tr>
+<tr>
+<td>en[i]</td><td>0</td><td>1</td><td>0</td><td>0</td><td>0</td>
+</tr>
+</tbody>
+</table>
+</center>
+
+Now we must resolve `dfs(4, 3)` and `dfs(5, 3)`. First, we resolve `dfs(4, 3)` by setting `st[4]` to the value of the timer, which is 3 in this case. Then, since `4` has no children, set `en[4]` to 3.
```

Comment:
```suggestion
Now we must resolve `dfs(4, 3)` and `dfs(5, 3)`. First, we resolve `dfs(4, 3)` by setting `st[4]` to the value of the timer (3 in this case) and incrementing the timer. Then, since `4` has no children, set `en[4]` to 3.
```